api/src/test/java/org/openmrs/util/OpenmrsUtilUnitTest.java

```diff
@@ -137,4 +147,330 @@ public void join_shouldReturnListElementsJoinedBySeparatorIfGivenSeparatorIsNull
+	@Test
+	public void isInAbsoluteNumericRange_shouldReturnTrueIfLowAbsoluteIsEqual() {
+
+		ConceptNumeric concept = new ConceptNumeric();
+		concept.setHiCritical(14.34d);
+		concept.setLowCritical(4.34d);
+
+		assertFalse(OpenmrsUtil.isInAbsoluteNumericRange(4.34f, concept));
+	}
+
+	@Test
+	public void isValidNumericValue_shouldReturnTrueIfHiAbsoluteIsNull() {
+
+		ConceptNumeric concept = new ConceptNumeric();
+		concept.setHiAbsolute(null);
+
+		assertTrue(OpenmrsUtil.isValidNumericValue(5.67f, concept));
+	}
+
+	@Test
+	public void isValidNumericValue_shouldReturnTrueIfLowAbsoluteIsNull() {
+
+		ConceptNumeric concept = new ConceptNumeric();
+		concept.setLowAbsolute(null);
+
+		assertTrue(OpenmrsUtil.isValidNumericValue(5.67f, concept));
+	}
+
+	@Test
+	public void isValidNumericValue_shouldReturnTrueIfValueIsInAbsoluteRange() {
+
+		ConceptNumeric concept = new ConceptNumeric();
+		concept.setHiAbsolute(10.34d);
+		concept.setLowAbsolute(3.67d);
+
+		assertTrue(OpenmrsUtil.isValidNumericValue(5.67f, concept));
+	}
+
+	@Test
+	public void isValidNumericValue_shouldReturnFalseIfLowAbsoluteIsNotInRange() {
+
+		ConceptNumeric concept = new ConceptNumeric();
+		concept.setHiAbsolute(10.34d);
+		concept.setLowAbsolute(6.67d);
+
+		assertFalse(OpenmrsUtil.isValidNumericValue(5.67f, concept));
+	}
+
+	@Test
+	public void isValidNumericValue_shouldReturnFalseIfHiAbsoluteIsNotInRange() {
+
+		ConceptNumeric concept = new ConceptNumeric();
+		concept.setHiAbsolute(10.34d);
+		concept.setLowAbsolute(6.67d);
+
+		assertFalse(OpenmrsUtil.isValidNumericValue(15.67f, concept));
+	}
+
+	@Test
+	public void isValidNumericValue_shouldReturnTrueIfHiAbsoluteIsEqual() {
+
+		ConceptNumeric concept = new ConceptNumeric();
+		concept.setHiCritical(4.34d);
+		concept.setLowCritical(3.67d);
+
+		assertFalse(OpenmrsUtil.isInAbsoluteNumericRange(4.34f, concept));
+	}
+
+	@Test
+	public void isValidNumericValue_shouldReturnTrueIfLowAbsoluteIsEqual() {
+
+		ConceptNumeric concept = new ConceptNumeric();
+		concept.setHiCritical(14.34d);
+		concept.setLowCritical(4.34d);
+
+		assertFalse(OpenmrsUtil.isInAbsoluteNumericRange(4.34f, concept));
+	}
+
+
+	@Test
+	public void isTestMode_shouldReturnTrueIfFunctionalTestModeIsEnabled() {
+
+		System.setProperty("FUNCTIONAL_TEST_MODE", "true");
+
+		assertTrue(OpenmrsUtil.isTestMode());
+	}
+
+	@Test
+	public void isTestMode_shouldReturnFalseIfFunctionalTestModeIsDisabled() {
+
+		System.setProperty("FUNCTIONAL_TEST_MODE", "false");
+
+		assertFalse(OpenmrsUtil.isTestMode());
+	}
+
+	@Test
+	public void compareLists_shouldPassIfMapOfDifferencesIsCorrect() {
+
+		LinkedList<Integer> origList = new LinkedList<>(Arrays.asList(1,2,3));
+		LinkedList<Integer> newList = new LinkedList<>(Arrays.asList(2,3,4));
+
+		Collection<Collection<Integer>> result = OpenmrsUtil.compareLists(origList, newList);
+		Iterator<Collection<Integer>> it = result.iterator();
+
+		assertThat(result.size(), is(2));
+		assertThat(it.next(), contains(4));
```

Comment:
does

`assertThat(result, contains(2, 4, 1));`

also work? if so please remove use of the iterator.

Reply:
@teleivo tried it. Since it is a nested collection doesn't work like that